public_html/wp-content/plugins/wordcamp-speaker-feedback/includes/class-rest-feedback-controller.php

```diff
@@ -0,0 +1,225 @@
+		) {
+			$feedback_author = array(
+				'name'  => $prepared_feedback['comment_author'],
+				'email' => $prepared_feedback['comment_author_email'],
+			);
+		} else {
+			return new WP_Error(
+				'rest_feedback_author_data_required',
+				__( 'Submitting feedback requires valid author name and email values.', 'wordcamporg' ),
+				array(
+					'status' => 400,
+				)
+			);
+		}
+
+		// We're not using the comment content field, but this also checks the length of other fields.
+		// The length of meta fields is checked separately. See `validate_feedback_meta()`.
+		$check_comment_lengths = wp_check_comment_data_max_lengths( $prepared_feedback );
+		if ( is_wp_error( $check_comment_lengths ) ) {
+			$error_code = $check_comment_lengths->get_error_code();
+			return new WP_Error(
+				$error_code,
+				__( 'Comment field exceeds maximum length allowed.' ),
+				array(
+					'status' => 400,
+				)
+			);
+		}
+
+		// We're not using this to set the status of the feedback comment, since it's always set to `hold`. However,
+		// this also checks for duplicates, flooding, and blacklisting.
+		$allowed = wp_allow_comment( $prepared_feedback, true );
+		if ( is_wp_error( $allowed ) ) {
+			$error_code    = $allowed->get_error_code();
+			$error_message = $allowed->get_error_message();
+
+			if ( 'comment_duplicate' === $error_code ) {
+				return new WP_Error( $error_code, $error_message, array( 'status' => 409 ) );
+			}
+
+			if ( 'comment_flood' === $error_code ) {
+				return new WP_Error( $error_code, $error_message, array( 'status' => 400 ) );
+			}
+
+			return $allowed;
+		}
+
+		$meta = validate_feedback_meta( $request['meta'] ?? array() );
+		if ( is_wp_error( $meta ) ) {
+			return $meta;
+		}
+
+		$comment_id = add_feedback( $prepared_feedback['comment_post_ID'], $feedback_author, $meta );
+
+		if ( false === $comment_id ) {
+			return new WP_Error(
+				'rest_feedback_creation_failed',
+				'Feedback submission failed.',
+				array(
+					'status' => 400,
+				)
+			);
+		}
+
+		$response = __( 'Success!', 'wordcamporg' );
+		$response = rest_ensure_response( $response );
+
+		$response->set_status( 201 );
+
+		return $response;
+	}
+
+	/**
+	 * Checks if a given request has access to create a feedback comment.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|bool True if the request has access to create items, error object otherwise.
+	 */
+	public function create_item_permissions_check( $request ) {
+		if ( empty( $request['post'] ) ) {
+			return new WP_Error(
+				'rest_feedback_invalid_post_id',
+				__( 'Sorry, you are not allowed to create this comment without a post.' ),
+				array(
+					'status' => 403,
+				)
+			);
+		}
+
+		$post = get_post( (int) $request['post'] );
+		if ( ! $post ) {
+			return new WP_Error(
+				'rest_feedback_invalid_post_id',
+				__( 'Sorry, you are not allowed to create this comment without a post.' ),
```

Comment:
I'm not sure why it's not recognizing your post ID. I just tried it again with Insomnia and it's still working for me...

> Should it also check for Session post type?

Yeah, I was thinking of including that as part of another validation function that would go where the `TODO` note currently is. I've been thinking I would add that as part of a separate PR for #340, but I guess that one is specifically about permissions for viewing/getting, so perhaps it would be better to do it here. Stay tuned.

Reply:
Add post type support and a check for it here: 2c9036af